### DISCUSSION TITLE
Putting Firefox in the Microsoft Store

### DISCUSSION THREAD

**Discussion by Justin-Vega** [mayrinck/FOSSonMicrosoftStore#1] [Requests] [answered]
Would it be out of the question to put Firefox in the Microsoft Store or would Mozilla be annoyed and immediately notice? If they were to quickly take it down, I don't see a point in trying. Otherwise, I really want it to happen! 🦊

Sorry if I'm doing something wrong here. I registered a GitHub account just to participate here. 🙂

**Comment by mayrinck** [comment, marked as answer]
Hey there, Justin! Welcome!

Actually, you did nothing wrong, the thing is that Microsoft Store only allows users to post browsers that are based on Microsoft Edge (the UWP version). Right now, there's no way to add a browser that is not made that way 😔



> EDIT: As pointed by @Justin-Vega, it is possible to upload browsers now if they are compatible with GECKO and Chromium and they can be only two versions prior to the version available from the official source. This allows browsers like Chromium and Forefox to be uploaded to the Store.

**Reply by Justin-Vega** [reply]
> Hey there, Justin! Welcome!
> 
> Actually, you did nothing wrong, the thing is that Microsoft Store only allows users to post browsers that are based on Microsoft Edge (the UWP version). Right now, there's no way to add a browser that is not made that way 😔

Thanks for the reply!

As far as I'm concerned, that isn't true anymore. Here is rule 10.2.1 of the Microsoft Store according to Microsoft.

_"Products that browse the web must use either the Chromium or the Gecko open source engine. To ensure compatibility and security of user experience they must be updated to be no older than within 2 major versions of those open source projects (for example, if the latest released major version of Chromium is 85, any browser based on Chromium must be on at least Chromium version 83 or higher) ,and known security issues must be patched in a more timely fashion. Any included private or proprietary components, or components not otherwise available under an open source license that affect compatibility of web site experience (such as codecs) shall be licensable on reasonable terms to other browser publishers to achieve compatibility. Compatibility and consistency of web site experience presented to browsers on the same engine shall be the primary test of meeting this engine consistency requirement (including publisher’s own sites). Existing browsers in the Windows Store may continue to use the EdgeHTML engine."_

**Reply by mayrinck** [reply]
Well, if it's true, I will try to upload the latest version of the Firefox there and see if it passes the certification, if so, we will get firefox there too. Thanks for the info!

**Reply by JuliaAgostino** [reply]
> Well, if it's true, I will try to upload the latest version of the Firefox there and see if it passes the certification, if so, we will get firefox there too. Thanks for the info!

Justin is right. See here: https://docs.microsoft.com/en-us/windows/uwp/publish/store-policies#102-security

**Reply by mayrinck** [reply]
I read the specific part he pointed, it's correct. I will try to upload the latest version and see what I got. The thing is that, for now, the main names of the browser are taken by someone else (maybe Mozilla itself?), then I will need to use an edited name... For now "Firefox Browser"

**Reply by JuliaAgostino** [reply]
> I read the specific part he pointed, it's correct. I will try to upload the latest version and see what I got. The thing is that, for now, the main names of the browser are taken by someone else (maybe Mozilla itself?), then I will need to use a close name...

Ah, okay. Good to know. Thanks! 🦊

**Reply by prakharb5** [reply]
There are news that Mozilla will be putting Firefox to MS store soon. 

Would there still be need to put Firefox in Store ourselves?

**Reply by JuliaAgostino** [reply]
> There are news that Mozilla will be putting Firefox to MS store soon.
> 
> Would there still be need to put Firefox in Store ourselves?

Yes, it's fine. If Mozilla does put Firefox in the Microsoft Store, we could always take ours down. 🙂

**Reply by mayrinck** [reply]
> 
> 
> > There are news that Mozilla will be putting Firefox to MS store soon.
> > Would there still be need to put Firefox in Store ourselves?
> 
> Yes, it's fine. If Mozilla does put Firefox in the Microsoft Store, we could always take ours down. 🙂

Exactly... As stated in the readme of the repo, if any developer publish their apps under their own account, we will pull out our version from the software 🙂. Also, the publishing may happen only  for Windows 11 devices, There's no reports or news stating that Microsoft Store will also change for Windows 10...

**Reply by prakharb5** [reply]
> There's no reports or news stating that Microsoft Store will also change for Windows 10...

There is, actually. Microsoft said they will be bringing the new Store to Windows 10 also.

> https://blogs.windows.com/windowsexperience/2021/06/24/building-a-new-open-microsoft-store-on-windows-11/

> Today, we are excited to introduce a new Microsoft Store coming soon to both Windows 11 **and Windows 10**. It will help you discover and install the best apps, games, movies and even special offers...

**Reply by JuliaAgostino** [reply]
> > > There are news that Mozilla will be putting Firefox to MS store soon.
> > > Would there still be need to put Firefox in Store ourselves?
> > 
> > 
> > Yes, it's fine. If Mozilla does put Firefox in the Microsoft Store, we could always take ours down. 🙂
> 
> Exactly... As stated in the readme of the repo, if any developer publish their apps under their own account, we will pull out our version from the software 🙂. Also, the publishing may happen only for Windows 11 devices, There's no reports or news stating that Microsoft Store will also change for Windows 10...

Mozilla Firefox is now officially in the Microsoft Store: https://www.microsoft.com/store/productId/9NZVDKPMR9RD

**Reply by prakharb5** [reply]
That too as an MSIX! Awesome! Didn't really expect that, ngl.

**Reply by JuliaAgostino** [reply]
> That too as an MSIX! Awesome! Didn't really expect that, ngl.

It's been expected for a while now: https://bugzilla.mozilla.org/show_bug.cgi?id=1549103#c5

**Comment by mayrinck** [comment]
Guys, Firefox is now under certification. I hope it passes this part (it's the hard one)

**Reply by JuliaAgostino** [reply]
My fingers are crossed! 🤞

News has already spread about your repository and how you're trying to submit Firefox to the Microsoft Store. Godspeed!

**Reply by mayrinck** [reply]
I hope everything goes well. There are more softwares I would love to see on the store right now, but I will stick to the ones listed up until now, convert/post them, and then try to upload a few others... Maybe some open-source games too...

**Reply by JuliaAgostino** [reply]
That would be great! There are plenty of Linux games featuring Tux that have long since been ported over to Windows. It would be great to see those in the store some day.

**Comment by mayrinck** [comment]
![image](https://user-images.githubusercontent.com/21313332/124637910-b6cc4880-de60-11eb-9a4d-61537e684edf.png)

Well, I guess Microsoft doesn't like firefox lol

**Reply by prakharb5** [reply]
"it must use the Windows Platform HTML and Javascript engines...."

They still want EdgeHTML? Havent they updated the requirements yet?

**Reply by mayrinck** [reply]
Sadly, It seems they really want the other browsers to be basically Classic Edge with another skin...

**Reply by Justin-Vega** [reply]
> Sadly, It seems they really want the other browsers to be basically Classic Edge with another skin...

That doesn't make sense to me. It says in the guidelines that they'll allow browsers that use the Chromium or Gecko engine... 🤔

**Reply by mayrinck** [reply]
> 
> 
> > Sadly, It seems they really want the other browsers to be basically Classic Edge with another skin...
> 
> That doesn't make sense to me. It says in the guidelines that they'll allow browsers that use the Chromium or Gecko engine... 🤔

What if this is only valid for Windows 11?

**Reply by JuliaAgostino** [reply]
> > > Sadly, It seems they really want the other browsers to be basically Classic Edge with another skin...
> > 
> > 
> > That doesn't make sense to me. It says in the guidelines that they'll allow browsers that use the Chromium or Gecko engine... 🤔
> 
> What if this is only valid for Windows 11?

Why would it only be valid for Windows 11?

**Reply by mayrinck** [reply]
Because they are changing the store, so maybe they have changed the policies before they can actually be valid through the store...

**Reply by JuliaAgostino** [reply]
> Because they are changing the store, so maybe they have changed the policies before they can actually be valid through the store...

That's odd. Why would they make those changes public before they're actually in effect instead of just telling people when they will go in effect or just not even posting them at all? I agree with @Justin-Vega, it doesn't make much sense (then again, Microsoft is terrible at communicating anything with their userbase, most notably the minimum requirements for Windows 11).
